lib/puppet/util/run_mode.rb

```diff
@@ -38,6 +38,10 @@ def log_dir
         "$vardir/log"
       end
 
+      def conf_dir
+        "$puppetdir/config"
+      end
+
```

Comment:
On Wed, Feb 04, 2015 at 10:08:09AM -0800, Eric Sorenson wrote:
> > @@ -38,6 +38,10 @@ def log_dir
> >          "$vardir/log"
> >        end
> >  
> > +      def conf_dir
> > +        "$puppetdir/config"
> > +      end
> > +
> 
> IIRC there is some extra magic about variable interpolation in settings, where only some settings which are "known" at this stage of the initialization can be interpolated... IIRC @jpartlow got down deep into this at one point, maybe as a result of https://tickets.puppetlabs.com/browse/PUP-1015  ?? 

This is easy: at any other location of the code the variable is correct,
except in the file I have added to the comment because it is called in
plain ruby, without any parser/etc option. It also means that *at that
particular stage* you can not overwrite --puppetdir with the command line.

> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/puppetlabs/puppet/pull/3556/files#r24104469

-- 
 (o-    Julien Pivotto
 //\    Open-Source Consultant
 V_/_   Inuits - https://www.inuits.eu